awx/main/analytics/collectors.py

```diff
@@ -0,0 +1,236 @@
+                'capacity': instance['capacity'],
+                'cpu': instance['cpu'],
+                'memory': instance['memory'],
+                'managed_by_policy': instance['managed_by_policy'],
+                }
+    return info
+
+
+@register('job_counts')
+def job_counts(since):
+    counts = {}
+    counts['total_jobs'] = models.UnifiedJob.objects.all().count()
+    counts['status'] = dict(models.UnifiedJob.objects.values_list('status').annotate(Count('status')))
+    counts['launch_type'] = dict(models.UnifiedJob.objects.values_list('launch_type').annotate(Count('launch_type')))
+    
+    return counts
+    
+    
+@register('job_instance_counts')
+def job_instance_counts(since):
+    counts = {}
+
+    job_types = models.UnifiedJob.objects.filter(created__gt=since).values_list(
+        'execution_node', 'launch_type').annotate(job_launch_type=Count('launch_type'))
+    for job in job_types:
+        counts.setdefault(job[0], {}).setdefault('status', {})[job[1]] = job[2]
+        
+    job_statuses = models.UnifiedJob.objects.filter(created__gt=since).values_list(
+        'execution_node', 'status').annotate(job_status=Count('status'))
+    for job in job_statuses:
+        counts.setdefault(job[0], {}).setdefault('launch_type', {})[job[1]] = job[2]
+    return counts
+
+
+# Copies Job Events from db to a .csv to be shipped
+def copy_tables(since, full_path):
+    def _copy_table(table, query, path):
+        events_file = os.path.join(path, table + '_table.csv')
+        try:
+            write_data = open(events_file, 'w', encoding='utf-8')
+            with connection.cursor() as cursor:
+                cursor.copy_expert(query, write_data)
+                write_data.close()
+        except Exception as e:
+            return e
+        return events_file
+
+    events_query = '''COPY (SELECT main_jobevent.id, 
+                              main_jobevent.created, 
+                              main_jobevent.event, 
+                              main_jobevent.failed, 
+                              main_jobevent.changed, 
+                              main_jobevent.playbook, 
+                              main_jobevent.role, 
+                              main_jobevent.job_id, 
+                              main_jobevent.host_id, 
+                              main_jobevent.host_name, 
+                              FROM main_jobevent 
+                              WHERE main_jobevent.created > {}
+                              ORDER BY main_jobevent.id ASC) to stdout'''.format(since.strftime("'%Y-%m-%d %H:%M:%S'"))
+    _copy_table(table='events', query=events_query, path=full_path)
+    
+    unified_job_query = '''COPY (SELECT DISTINCT main_unifiedjob.id, 
+                                 main_unifiedjob.polymorphic_ctype_id,
+                                 django_content_type.model,
+                                 main_unifiedjob.created, 
+                                 main_unifiedjob.description, 
```

Comment:
we can probably lose the following fields:
* description
* celery_task_id
* created_by_id
* old_pk